src/python/WMCore/WMSpec/WMWorkload.py

```diff
@@ -881,6 +930,8 @@ def setLumiList(self, lumiLists):
         return
 
     def setTaskProperties(self, requestArgs):
+        # FIXME (Alan): I don't think it works, given that the assignment
+        # parameters never have the TaskChain parameter...
```

Comment:
I think it is intended to use in the case some one (dataops, unified) wants to update Task specific properties in assignment. 'i.e' {'TaskName': "FristTask", "KeepOut": True}. But I am not sure we have the use case for that. If we have the use case. We can just fix it (maybe in other patch - by checking 'TaskName' key  instead of "TaskChain" key). If not, we can just remove it.

Reply:
I see what you mean. At some point you changed TaskChain such that it would accept updates in two formats, but only one got used and I think I ended up breaking those 2 formats with some changes. This is how Ops assign/update TaskChain arguments:
https://github.com/dmwm/WMCore/blob/master/test/data/ReqMgr/requests/Integration/TaskChain_Prod.json#L3

I think we should accept only one format, less work and easier to maintain the code. Going to the meeting now :)

Reply:
I agree, just remove it then. Thanks Alan.

Reply:
I still suspect the format you had in mind would be easier to deal with, also for the update functions and etc :-)
Changing the code to support updates such `{'Task1': {'EventsPerJob': 1000}, 'Task2': {'KeepOutput': False}...}` wouldn't be simple though, maybe we can have a chat after CompOps meeting.